triggering_test.go

```diff
@@ -587,20 +636,40 @@ func BenchmarkLevelTrigger0TriggersOpsAreSamples(b *testing.B) {
 
 }
 
+func almostEqual(a, b, threshold float64) bool {
+	return math.Abs(a-b) <= threshold
+}
+
 func TestKinkModel(t *testing.T) {
 	xdata := []float64{0, 1, 2, 3, 4, 5, 6, 7}
 	ydata := []float64{0, 0, 0, 0, 1, 2, 3, 4}
-	ymodel, a, b, c, X2, err := kinkModelResult(3, xdata, ydata)
-	if a != 0 || b != 0 || c != 1 || X2 != 0 || err != nil {
```

Comment:
This test was failing ONLY on my new M1 Mac, because it was testing for exact floating-point equality. Always a bad idea, but only now did it cause any problems. Anyway, I replaced that with tests for approximate equality `almostEqual(...)`, and now it passes on all computers.